utils/yoke/design_doc.md

```diff
@@ -1,20 +1,18 @@
-# Yoke: Lifetime Erasure for Rust
+# Yoke: Self-Referential Borrowing for Rust
```

Comment:
@sffc I'd really prefer to call this "lifetime erasure", "self referential borrowing" is a huge field in Rust with many crates already. When explaining what `yoke` does to other rustaceans I've found "lifetime erasure" to be quite clear since it makes a clear analogy with "type erasure" (`dyn`)

Reply:
I see lifetime erasure as being a means to an end.  Yoke applies its lifetime erasure to a very narrow use case, which is self-references.  I took another stab at the headline which mentions both things.

Reply:
I still kinda feel like the relationship is inverted: the point of Yoke is lifetime erasure, and it accomplishes that by self referential borrowing. There's no way lifetime erasure can work _without_ self referential borrowing as well.

Reply:
Ultimately, to me, "self referential borrowing" does not tell me much because it covers so many things. "lifetime erasure" tells me exactly what this crate is useful for: I want to turn a compile time lifetime into an erased runtime one.

Reply:
Like, to be clear, by "lifetime erasure" I'm not talking about the fact that yoke replaces lifetimes with static as a means to an end.

I'm making an analogy against type erasure: turn a compile time construct into a runtime one.

Reply:
Let me write it out in the doc